source/grid/manifold_lib.cc

```diff
@@ -1321,6 +1321,26 @@ FunctionManifold<dim, spacedim, chartdim>::FunctionManifold(
 
 
 
+template <int dim, int spacedim, int chartdim>
+FunctionManifold<dim, spacedim, chartdim>::FunctionManifold(
+  std::unique_ptr<Function<chartdim>> &push_forward_function,
+  std::unique_ptr<Function<spacedim>> &pull_back_function,
+  const Tensor<1, chartdim> &          periodicity,
+  const double                         tolerance)
+  : ChartManifold<dim, spacedim, chartdim>(periodicity)
+  , const_map()
+  , push_forward_function(push_forward_function.release())
+  , pull_back_function(pull_back_function.release())
```

Comment:
You can just copy them here -- no need to write `.release()`.

Reply:
Hm... If I don't release them, won't they be destroyed then when unique_ptr goes out of scope? Or what do you mean by copying?

Reply:
If you write `pull_back_function(pull_back_function)` (with the second occurrence of the word being the function argument), then you are invoking the move constructor of `std::unique_ptr` since there is no copy constructor of that class. The move constructor moves ownership from the source to the destination objects. That's exactly what you are doing here, just in a more complicated way :-)

Reply:
The member variable is a `SmartPointer`, not a `std::unique_ptr`. I guess this is correct as is.

Reply:
Oh oh, yes, then that is correct.

Reply:
This is then ready to merge, isn't?